tls/s2n_connection.c

```diff
@@ -100,6 +101,7 @@ struct s2n_connection *s2n_connection_new(s2n_mode mode)
     GUARD_PTR(s2n_stuffer_growable_alloc(&conn->in, 0));
     GUARD_PTR(s2n_stuffer_growable_alloc(&conn->handshake.io, 0));
     GUARD_PTR(s2n_connection_wipe(conn));
+    GUARD_PTR(s2n_socket_snapshot(conn));
```

Comment:
I don't this works if the s2n_connection handle is being reused via s2n_connection_wipe. We'd only save the socket options on the initial allocation, but not for subsequent use after wiping. Maybe we could save the options in s2n_connection_set_fd?